### DISCUSSION TITLE
BUG: QuotaExceededError: Failed to execute 'setItem' on 'Storage'

### DISCUSSION THREAD

**Discussion by DorianDaSilva** [WalletConnect/walletconnect-monorepo#3566] [Developer Support]

**Comment by SergeyVolynkin** [comment]
@glitch-txs when you have time, could you please clarify why this can happen and if this type of error can affect wallet connection and signing for production users 🤔? We've got few errors like this in production

**Reply by glitch-txs** [reply]
Hi, this happens when the localStorage is full

**Reply by SergeyVolynkin** [reply]
Oh, I see, thank you -- at first I thought it some sort of rate-limiting on WalletConnect cloud side 🤔-- is it always just related to localStorage? @glitch-txs 

Still not sure how this can happen for the unmodified WalletConnect web3 modal usage 🤔 is there some memleak issue open for the current version of WalletConnect modal?

**Reply by luccamordente** [reply]
> Hi, this happens when the localStorage is full

@glitch-txs `wc@2:core:0.3//history` seems to hoard quite a lot of records. I've been using WalletConnect for just a few days and can see almost 3,000 records in there. Is that normal? We use localStorage for other things in our app and we can't risk running out of space there that easy.

**Reply by SergeyVolynkin** [reply]
Good catch, @luccamordente, Could you please setup a new discussion in https://github.com/orgs/WalletConnect/discussions/categories/web3inbox-sdk-support so we can get insights if this extra logging to localStorage can be reduced somehow (or will be fixed in future WalletConnect releases). Maybe it will be easier that way to get expanded @glitch-txs response on this

**Reply by franfernandez20** [reply]
Hello. I'm also having problems when using large messages through WalletConnect. After two or three of these significant transactions, I noticed this problem. It would be helpful to add some configuration to set the maximum size usable on localStorage.  

Can I ask for a proper way to remove only this huge transaction from history (and other places) once I process it? 
Thank you
